madmom/ml/hmm.pyx

```diff
@@ -394,6 +394,38 @@ class HiddenMarkovModel(object):
             raise ValueError('Initial distribution is not a probability '
                              'distribution.')
         self.initial_distribution = initial_distribution
+        # attributes needed for stateful processing (i.e. forward_step())
+        self._prev = self.initial_distribution.copy()
+        self._cur = np.zeros_like(self._prev)
```

Comment:
Yes, calling `reset()` does the same.

IMHO, we should get rid of `forward_generator` altogether in the long run, since it basically does the same as `forward()` with `reset=False` called with individual observations. What the current implementation of `forward()` doesn't support is handling of `block_size` other than 1. Thus I did not remove it for now.